lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1703,12 +1703,12 @@
                         platform.raiseNeedObjectOfType("DateTimeFormat.prototype.format", "DateTimeFormat");
                     }
 
-                    if (isNaN(date) || !isFinite(date)) {
+                    if (date !== undefined && (isNaN(date) || !isFinite(date))) {
```

Comment:
>isNaN(date) || !isFinite(date) [](start = 47, length = 30)

Do these checks do the wrong thing is typeof date is not 
'number'? I'm having trouble parsing this validation logic.

Maybe I'm missing something but I think the cases go something like this

if date === undefined -> raise
if !date (otherwise falsy) -> date = epoch
if typeof date !== 'number' -> raise
if typeof date === 'number' && (isNAN(date) or !isFinite(date)) -> raise

Reply:
Date doesn't have to be a number, as its coerced with the ecma262 `ToNumber` abstract operation regardless. Our JS differs from spec heavily here but https://tc39.github.io/ecma402/#sec-datetime-format-functions basically shows what to do (FormatDateTime calls PartitionDateTimePattern which checks for the date being finite. I honestly don't know where the NaN check comes in, but all browsers agree that NaN is bad)

Reply:
> don't know where the NaN check comes in

My spec read:

https://tc39.github.io/ecma402/#sec-partitiondatetimepattern step 1
via https://tc39.github.io/ecma402/#sec-formatdatetime step 1

> 1. If x is not a finite Number, throw a RangeError exception

`not a finite Number` is something that should probably be made more clear by the spec... closest thing I can find is:

https://tc39.github.io/ecma262/#sec-terms-and-definitions-number-type 

> 4.3.21 Number type
> set of all possible Number values including the special “Not-a-Number” (NaN) value, positive infinity, and negative infinity

I think I'd say that: `x` is `not a finite number` implies `[-Infinity, Infinity, NaN].includes(x) === true` (though this is not formalized in the spec) because "Not-a-Number" is `not a finite number`.

> 4.3.24 NaN
> number value that is an IEEE 754-2008 “Not-a-Number” value

Reply:
From IRC:

```
<•Bakkot> ah: https://tc39.github.io/ecma262/#sec-ecmascript-language-types-number-type
5:57 PM D<dilijev> Doug Ilijev i.e. "The other 18437736874454810624 (that is, 264-253) values are called the finite numbers."
5:57 PM "The Number type has exactly 18437736874454810627 (that is, 264-253+3) values"
5:57 PM and the other 3 are [Infinity, -Infinity, NaN]
```